sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -659,12 +670,23 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
         AttributeReference("value", StringType)()), true)
     }
 
-    // Create the transform.
+    val plan = visitCommonSelectQueryClausePlan(
+      relation,
+      lateralView,
+      transformClause.namedExpressionSeq,
+      whereClause,
+      aggregationClause,
+      havingClause,
+      windowClause,
+      isDistinct = false)
+
+    // Create the transform, here we pass UnresolvedStart as ScriptTransform's input.
+    // In analyzer after child plan is resolved, we resolve UnresolvedStart as child's output.
     ScriptTransformation(
-      expressions,
+      Seq(UnresolvedStar(None)),
```

Comment:
hm, on second thought, we cannot remove this param `input` from `ScriptTransformation` in this PR? Since the input exprs of the current `ScriptTransformation` implementaiton always coms from child's output, IIUC we don't need this param anymore?

Reply:
> hm, on second thought, we cannot remove this param `input` from `ScriptTransformation` in this PR? Since the input exprs of the current `ScriptTransformation` implementaiton always coms from child's output, IIUC we don't need this param anymore?

It looks like this. We can replace `input` with `child. output` directly. That's a really nice suggestion since current way(converting LogicalPlan it looks a little weird). If we remove this `input` parameter and add correct  comment.
The whole process looks more natural.  I have tried it in local, a new big diff. How about a new ticket about refactor this?

Reply:
Nice. Just removing the param causes a big diff? I'd like to remove[ the current the weird Analyzer code ](https://github.com/apache/spark/pull/29087/files#diff-ed19f376a63eba52eea59ca71f3355d4495fad4fad4db9a3324aade0d4986a47R1520-R1526) to handle the unresolved star in this PR though.

Reply:
> Nice. Just removing the param causes a big diff? I'd like to remove[ the current the weird Analyzer code ](https://github.com/apache/spark/pull/29087/files#diff-ed19f376a63eba52eea59ca71f3355d4495fad4fad4db9a3324aade0d4986a47R1520-R1526) to handle the unresolved star in this PR though.

Yea, will change a lot of file.  Compared to the origin code here(the current the weird Analyzer code) ，current change seem s not so weird, but  for the whole process, really weird.

Create a ticket for this https://issues.apache.org/jira/browse/SPARK-34035
![image](https://user-images.githubusercontent.com/46485123/103833742-afe49900-50bc-11eb-979a-b610582383d0.png)

Reply:
> Yea, will change a lot of file. 

How many changed lines of codes there?

Reply:
> > Yea, will change a lot of file.
> 
> How many changed lines of codes there?

Nearly 100 lines, most of changes are about UT

Reply:
okay, I don't have a strong opinion on it, so please follow other reviewer's comment.

Reply:
> okay, I don't have a strong opinion on it, so please follow other reviewer's comment.

Yea